### PR TITLE
Allow sorting of cards

### PR DESCRIPTION
**Pull request by anselm92**
Cards that are of a different type can be drag and dropped
up and down the list to change their location (persistent).

## Issue

This fixes the following issue(s): #228

## Screenshot

screenshot_20180118-004710

## Why this is useful for all students

Users can change positions of cards according to their needs

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/cards/generic/Card.java [hunk 1]
```diff
@@ -209,6 +210,28 @@ public void hideAlways() {
...
+        e.apply();
+    }
+
+    @Override
+    public int compareTo(@NonNull Card card) {
+        if(card.getPosition()>getPosition()) {
+            return -1;
+        } else {
+            return 1;
+        }
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/cards/generic/Card.java]: I'd do an `Integer.compare(card.getPosition(), getPosition())` or vice versa

**Comment by anselm92** [comment]
Ok (y) 
I've two questions though:
1) On restore cards, should the position be reset too?
2) Should sticky cards still be on top or moveable too?

**Comment by psukys** [comment]
@anselm92 if restore does some kind of resetting, then I think the position should change as well.

**Comment by kordianbruck** [comment]
@anselm92 we only have one sticky card, so make these persistent at the top. Once you dismiss them, they usually never reappear.

How does this interact with swiping the cards right/left? Is that still possible?

**Comment by anselm92** [comment]
@kordianbruck 
alright.
yes you can still dismiss cards.

**Comment by pfent** [comment]
@kordianbruck This one can also be merged

**Pull request merged**